src/classes/keyframe_scaler.py

```diff
@@ -0,0 +1,87 @@
+"""
+ @file
+ @brief Process project data, scaling keyframe X coordinates by the given factor
+ @author Noah Figg <eggmunkee@hotmail.com>
+ @author Jonathan Thomas <jonathan@openshot.org>
+ @author Olivier Girard <eolinwen@gmail.com>
+ @author FeRD (Frank Dana) <ferdnyc@gmail.com>
+
+ @section LICENSE
+
+ Copyright (c) 2008-2020 OpenShot Studios, LLC
+ (http://www.openshotstudios.com). This file is part of
+ OpenShot Video Editor (http://www.openshot.org), an open-source project
+ dedicated to delivering high quality video editing and animation solutions
+ to the world.
+
+ OpenShot Video Editor is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OpenShot Video Editor is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with OpenShot Library.  If not, see <http://www.gnu.org/licenses/>.
+ """
+
+
+class KeyframeScaler:
+    """This factory class produces scaler objects which, when called,
+    will apply the assigned scaling factor to the keyframe points
+    in a project data dictionary. Keyframe X coordinate values are
+    multiplied by the scaling factor, except X=1 (because the first
+    frame never changes)"""
+
+    def _scale_value(self, value: float) -> int:
```

Comment:
Interesting, I have not seen type hints before in Python. Pretty cool. So, in this case, we are passing in a float, but returning an int? Is that by design?

Reply:
Oh I see, this is only called with X values (which are of type INT), representing frame #'s

Reply:
Should we rename this to be more specific to INT or FRAME NUMBER or X?

Reply:
That's actually a good point. The old code produced int values because it did the same thing:
```py
return round(original_value * scale_factor)
```
...But even X coordinate values _technically_ are floats, so maybe the function should convert the new rounded integer values to float, and return those instead.

As far as the name, yeah `_scale_x_value()` could work, sure... The keyframe scaler's entire function is to scale X values, but that's no reason _not_ to make it explicit, and therefore clearer, still.

Reply:
I changed it to `_scale_x_value()`, but left the `int` return since it matches the old code.